mantidproject/mantid
# GUI Restructuring 13: Update table with output values

***PR ~~#25843~~ should be merged first.*** In the meantime this can be reviewed by comparing against [that branch](https://github.com/gemmaguest/mantid/compare/23027_refl_gui_live_monitor_settings...23027_refl_gui_update_output_values)

**Description of work.**

Ongoing work to integrate the `JobTreeView` into the `ISIS Reflectometry` GUI. This PR is being merged to a feature branch because not all functionality is implemented yet.

This PR updates the Q min/max/step values from the algorithm outputs if those cells did not contain user-specified input values. This is the same as the original GUI, however an addition in this PR is that the output values are greyed out to distinguish them from input values. If the user subsequently edits the value then it will be shown in black to indicate it is a user input rather than an output.

**To test:**

- Open the `ISIS Reflectometry` interface and set the instrument to `INTER`
- Enter some run numbers and angles e.g. as per the following image, and the process everything
![image](https://user-images.githubusercontent.com/12895056/59209488-5300c400-8ba3-11e9-9310-815e485c0fd9.png)
- Try editing the q min/max/step and check that the output workspaces are cropped to the min/max and rebinned to the q-step appropriately.
- Check that the user-entered values are shown in black and the other cells are populated from the outputs and greyed out.
- Try deleting a value you entered and reprocessing and check that it returns to the output value from the algorithm and is greyed out.

Refs #23027

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
